### ISSUE TITLE
Extracting Vuforia source code from plugin (to stay abreast of the Vuforia license)

### ISSUE THREAD

**Issue by nleclerc** [mattrayner/cordova-plugin-vuforia#18]
I've noticed that the `build/include/QCAR` and `src/vuforia/include/QCAR` contain the exact same files.
I recon that the build folder could be remove, what do you think ?

**Comment by mattrayner** [comment]
Hi @nleclerc,
 You're quite possibly right. I think this has happened because the Android and iOS sections of the plugins were developed within the same team but by two different people.

My concern with just completely ripping out the build folder at the moment is that we **may** break some people's set ups. Our current iOS instructions involve pointing XCode specifically at the build folder.

We'd be interested to see how you'd like to see it working?

_My_ initial thoughts are that we could (to save backwards compatibility) either move the Vuforia code inside of `src/vuforia/*` into `build` or, as you say, delete the build folder all together.

@dsgriffin, @daniel-jackson95 - Would love you get your input on this too?

**Comment by daniel-jackson95** [comment]
I think that moving `src/vuforia/*` into the `build` folder is probably the best solution.
We just spoke about removing the `build` folder, but that could possibly break any previous projects where xcode links to files in that directory (and then stops working).

**Comment by nleclerc** [comment]
Why not just declare the source files so that they are properly added to the code project ?
There shouldn't be the need for any manual configuration really.

I think the ideal solution would be to move vuforia files outside of this plugin.
I'm not even sure the licensing allows redistribution of the sdk.

**Comment by mattrayner** [comment]
Well, with Android that works fine currently - it's all added automatically - the issue is with iOS and Xcode... Having read the developer agreement here:

**Comment by nleclerc** [comment]
Well i'm not really sure but for what i understand, the dev agreement allows distribution of the engine so that the software can work but it doesn't seem to cover the case of the associated source and header files.

The only way to be sure would be to contact vuforia and ask.
As recent court cases have proven, it might not be that simple :-)

I looked into making a plugin that automatically downloads the sdk from the vuforia website.
I think it can be done but it's hampered by the download process on the official website (need to send multipart POST request to download).
I haven't had time to get further into it but it could be a nice (and legit) solution.

**Comment by mattrayner** [comment]
We think we've come up with a slightly different solution to prevent us falling foul of the Vuforia license.

Our current thinking is you download the Android and iOS SDK zips from Vuforia (thus complying to their license) and then we run a script to move them into a location Cordova can get at them.

Our script currently looks like this:

```ruby
puts 'Checking for files in \'vuforia\' folder'
@android_zip = Dir['vuforia/vuforia-sdk-android-*.zip']
@ios_zip = Dir['vuforia/vuforia-sdk-ios-*.zip']

puts 'Found Android sdk zip' unless @android_zip.empty?
puts 'Found iOS sdk zip' unless @ios_zip.empty?

abort('Unable to find Android or iOS SDK zips inside \'vuforia\' folder. Expected files with format \'vuforia-sdk-PLATFORM-*.zip\' to exist.') if @android_zip.empty? and @ios_zip.empty?

puts 'Creating build directory'
system('mkdir -p build/include build/lib')

unless @android_zip.empty?
    puts '----------'
    puts 'Processing Android'

    puts 'Creating Directories'
    system('mkdir -p build/java vuforia/android')

    puts 'Extracting Android SDK'
    system("tar -xvf #{@android_zip.first} --directory vuforia/android")

    puts 'Copying Build Assets'
    system('cp -r vuforia/android/build/include/ build/include/')
    system('cp -r vuforia/android/build/java/ build/java/')
    system('cp -r vuforia/android/build/lib/ build/lib/')

    puts 'Cleaning Android'
    system('rm -r vuforia/android')
end

unless @ios_zip.empty?
    puts '----------'
    puts 'Processing iOS'

    puts 'Creating Directories'
    system('mkdir -p vuforia/ios')

    puts 'Extracting iOS SDK'
    system("tar -xvf #{@ios_zip.first} --directory vuforia/ios")

    puts 'Copying Build Assets'
    system('cp -r vuforia/android/build/include/ build/include/')
    system('cp -r vuforia/ios/build/lib/ build/lib/')

    puts 'Cleaning iOS'
    system('rm -r vuforia/ios')
end

puts 'Done.'
```

(this is obviously REALLY early) and we haven't even begun to test it yet

**Comment by mattrayner** [comment]
This will be a breaking change, so we will be moving this to a 2.0 milestone.

**Comment by mattrayner** [comment]
It looks like you can only download Vuforia V5 from the developer site... The current plugin runs on 4 - it looks like a change to 5 would require a re-write of the plugin itself unless someone can find the place where 4 lives?

**Comment by mattrayner** [comment]
I've temporarily set this as 'wont fix' - we will update this at a later date. We will not consider this as a pressing issue right now though

**Comment by mattrayner** [comment]
I've set this to a 4.0.0 ticket - this is so we don't just forget about it

Well, specifically so **I** don't forget about it (brain like a sieve)

**Comment by nleclerc** [comment]
Seems like you need to be logged in to download the SDK (i'll never understand why people gate those). So unless we create a dummy account to allow direct download, i believe the magic won't happen :-(

**Comment by nleclerc** [comment]
I believe we could do a plugin containing only the sdk, that would be easier to manager if they somehow decide to crack down on us and we need a plan b.

**Comment by mattrayner** [comment]
that is a good idea @nleclerc - I'll look into this tonight as an option.

**Comment by mattrayner** [comment]
I've began extracting the SDK out - it can be found here:

**Comment by mattrayner** [comment]
This is now addressed in #95
